pinot-controller/src/test/java/org/apache/pinot/controller/helix/core/realtime/PinotLLCRealtimeSegmentManagerTest.java

```diff
@@ -1035,6 +1036,13 @@ public void testUploadToSegmentStore()
 
     // Verify the result
     segmentManager.uploadToDeepStoreIfMissing(segmentManager._tableConfig, segmentsZKMetadata);
+
+    // Block until all tasks have been able to complete
+    ThreadPoolExecutor deepStoreUploadExecutor = segmentManager.getDeepStoreUploadExecutor();
+    while (deepStoreUploadExecutor.getActiveCount() > 0) {
```

Comment:
Seems active count might not always be accurate (e.g. task not scheduled yet). Shall we wait on downloadUrl being set?

Reply:
Changed to use pending segment set size, as waiting on downloadUrl makes it difficult to correctly test the error cases where we assert downloadUrl isn't set